src/app/pages/component-viewer/component-viewer.ts

```diff
@@ -17,12 +17,17 @@ export class ComponentViewer {
   componentDocItem: DocItem;
 
   constructor(private _route: ActivatedRoute,
+              private router: Router,
               public _componentPageTitle: ComponentPageTitle,
               public docItems: DocumentationItems) {
-    _route.params.subscribe(p => {
-      this.componentDocItem = docItems.getItemById(p['id']);
+    const id = this._route.snapshot.paramMap.get('id');
+    this.componentDocItem = docItems.getItemById(id);
+
+    if (this.componentDocItem) {
       this._componentPageTitle.title = `${this.componentDocItem.name}`;
```

Comment:
I don't know if there's any reason to wrap `this.componentDocItem.name` with template literal, but I kept it as it was.